src/Module/Item/Ignored.php

```diff
@@ -43,10 +44,20 @@ public static function rawContent()
 		// Numeric values are needed for the json output further below
 		$ignored = !empty($thread['ignored']) ? 0 : 1;
 
-		if (!empty($thread['uid']) && $thread['uid'] != 0) {
-			$dba->update('thread', ['ignored' => $ignored], ['iid' => $message_id]);
-		} else {
-			$dba->update('user-item', ['ignored' => $ignored], ['iid' => $message_id, 'uid' => local_user()], true);
+		switch ($thread['uid'] ?? 0) {
+			// if the thread is from the current user
+			case local_user():
+				$dba->update('thread', ['ignored' => $ignored], ['iid' => $message_id]);
+				break;
+			// Empty or 0 (null will get transformed to 0) => it's a public post
+			case 0:
+			case '':
```

Comment:
There's no reason for this line, switch comparison isn't strict that I know of, and all threads have a uid, 0 or >0.

Reply:
So there's no possibility that an empty string is returned by some of the inner functions? Normaly I don't trust not typesafed content. That's why I try to guess any possible content. But if the comparison doesn't care about it, it's fine for me too :)

Reply:
I've never heard `switch` was type-safe.